opm/simulators/linalg/bda/cusparseSolverBackend.cu

```diff
@@ -227,8 +227,8 @@ void cusparseSolverBackend<block_size>::initialize(int N, int nnz, int dim) {
     cudaMalloc((void**)&d_t, sizeof(double) * N);
     cudaMalloc((void**)&d_v, sizeof(double) * N);
     cudaMalloc((void**)&d_bVals, sizeof(double) * nnz);
-    cudaMalloc((void**)&d_bCols, sizeof(double) * nnz);
-    cudaMalloc((void**)&d_bRows, sizeof(double) * (Nb + 1));
+    cudaMalloc((void**)&d_bCols, sizeof(int) * nnzb);
+    cudaMalloc((void**)&d_bRows, sizeof(int) * (Nb + 1));
```

Comment:
Would it be possible to write this as
```C++
cudaMalloc((void**)&d_bCols, nnzb * sizeof *d_bCols);
cudaMalloc((void**)&d_bRows, (Nb + 1) * sizeof *d_bRows);
```
instead?  Ignoring the fact that `(void**) &some_pointer` is a constraint violation unless `some_pointer` is the exact type `void*`, I always prefer to use `sizeof object` instead of `sizeof(type)` when I have to use a `malloc`-like interface.

Reply:
I agree, I plan a separate PR regarding this to clearly separate this issue. Otherwise, there are a lot of places where this needs to be changed. Would that be an OK solution for you?

Reply:
> I plan a separate PR regarding this to clearly separate this issue. [...] Would that be an OK solution for you?

Perfect!